src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -1238,6 +1238,35 @@ internal bool HasRequiresLocationAttribute(EntityHandle token)
         }
 
 #nullable enable
+        /// <summary>
+        /// Indicates whether the first attribute should be prioritized over the second one.
+        /// Same order of priority as
+        ///   <see cref="TryGetDeprecatedOrExperimentalOrObsoleteAttribute(EntityHandle, IAttributeNamedArgumentDecoder, bool, bool)"/>
+        /// </summary>
+        internal static bool IsMoreImportantObsoleteKind(ObsoleteAttributeKind oldKind, ObsoleteAttributeKind newKind)
```

Comment:
>IsMoreImportantObsoleteKind [](http://example.com/codeflow?start=29&length=27)

Could you clarify the intended behavior for this helper? Is this as simple as a priority order between the attributes? If so, I think implementation could be more direct and much simpler to follow. Something like get priority for each kind, compare priorities.

Reply:
It looks like here is the list of kinds in priority order according to ```TryGetDeprecatedOrExperimentalOrObsoleteAttribute``` (from highest to lowest):
- Deprecated
- Obsolete
- WindowsExperimental
- Experimental